beacon_chain/types/src/validator_registry.rs

```diff
@@ -0,0 +1,171 @@
+/// Contains logic to manipulate a `&[ValidatorRecord]`.
+/// For now, we avoid defining a newtype and just have flat functions here.
+use super::validator_record::*;
+
+/// Given an indexed sequence of `validators`, return the indices corresponding to validators that are active at `slot`.
+pub fn get_active_validator_indices(validators: &[ValidatorRecord], slot: u64) -> Vec<usize> {
+    validators
+        .iter()
+        .enumerate()
+        .filter_map(|(index, validator)| {
+            if validator.is_active_at(slot) {
+                Some(index)
+            } else {
+                None
```

Comment:
This could be a consensus issues if other clients don't follow suit.

(Repeat comment)

Reply:
I may misunderstand you but I think this is a different instance than the other use of `Option<T>`.

Here, we just use the variant of `Option` to signal to `filter_map` which values to keep. The return value is just a `Vec<usize>` containing the active indices.

I was thinking about this and the place where I do see potential for trouble is w/ the order of these indices. I suppose we implicitly go over insertion order of the `ValidatorRecords` every time we traverse them in the spec. If that order ever deviates then it would quickly lead to consensus failure.

The applicable point here is that this function does provide some order, but makes no guarantees in the interface. And regarding the above point, it is actually up to the caller to make sure to keep `validators` in insertion order. As long as other parts of the repo follow the spec it should "all just work", but idk if you have any thoughts around the safety here (safety like how safe the code is to use correctly, not consensus safety)

Reply:
Oh you're right, I am mistaken. Sorry!

Good points regarding insertion order. I can't think of any good solution this, apart from maybe creating some `OrderedValiators` newtype, but I feel like that's a bit heavy handed. I can't imagine it's likely for us to have un-ordered validator arrays floating around. Maybe we can just keep this in the back of our minds and implement something if an issue appears.